server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/processors/FunctionReferenceEnricher.java

```diff
@@ -28,12 +28,16 @@ public class FunctionReferenceEnricher implements Processor<FunctionReference> {
 
   @Override
   public void accept(FunctionReference node, ProcessingContext processingContext) {
-    SymbolAccumulatorService.FunctionInfo fi = symbolAccumulatorService.getFunctionReference(node.getName());
-    if (fi == null)
-      return;
-
-    node.setDefinitions(fi.getDefinition());
-    node.setUsages(fi.getReferences());
+    if (node.isFunctionPrefixed()) {
+      node.getProgram()
+          .map(
+              programNode ->
+                  symbolAccumulatorService.getFunctionReference(node.getName(), programNode))
```

Comment:
This still does not find the function if it is not declared in the repository, right?